test/integration/targets/connection_ssh/test_ssh_askpass.yml

```
@@ -0,0 +1,57 @@
+- hosts: all
+  tasks:
+    - import_role:
+        role: setup_test_user
+
+    - command: grep -r ^PasswordAuthentication /etc/ssh/
+      register: sshd_conf
+      ignore_errors: true
+
+    - lineinfile:
+        path: '{{ sshd_conf_file }}'
+        regexp: '^PasswordAuthentication'
+        line: PasswordAuthentication yes
+      vars:
+        sshd_conf_file: >-
+          {% if sshd_conf is successful and sshd_conf.stdout_lines|length == 1 %}{{ sshd_conf.stdout|split(':')|first }}{% else %}/etc/ssh/sshd_config{% endif %}
+
+    - when: ansible_facts.system == 'Darwin'
+      block:
+        - command: launchctl stop com.openssh.sshd
+        - command: launchctl start com.openssh.sshd
+
+    - service:
+        name: ssh{{ '' if ansible_facts.os_family == 'Debian' else 'd' }}
+        state: restarted
+      when: ansible_facts.system != 'Darwin'
+
+    - command:
+        argv:
+          - ansible
+          - localhost
```

This should connect to `testhost`, not `localhost`, since they're not always the same.

```suggestion
          - testhost
```

---

I actually had that, by using `../../inventory`, the problem is that the inventory doesn't specify `ansible_host=127.0.0.1`, and as such `testhost` isn't resolvable:

```
ssh: Could not resolve hostname testhost: Name or service not known
```

There is additionally a `test_connection.inventory` that provides 2 hosts, which are for pipelining tests that do set `ansible_host=localhost`.

---

Ah yes, I'd forgotten that connection tests run in the controller context, so `testhost` is `localhost` in this case.